tpchgen/src/generators.rs

```diff
@@ -8,24 +8,23 @@ use crate::random::RandomBoundedLong;
 use crate::random::RandomPhoneNumber;
 use crate::random::RowRandomInt;
 use crate::text::TextPool;
-use std::sync::Arc;
 
 use crate::dates::{GenerateUtils, TPCHDate};
 use crate::random::{RandomBoundedInt, RandomString, RandomStringSequence, RandomText};
 
 /// Generator for Nation table data
-pub struct NationGenerator {
+pub struct NationGenerator<'a> {
```

Comment:
I think making the generation multi-threaded will be significantly harder using the approach in https://github.com/clflushopt/tpchgen-rs/pull/34 if there are lifetimes on the generators

However it seems to me that all the distributions are effectively `'&static` (don't change) so maybe instead of putting lifetimes on the generators, we could just make all the references to `TextPool` `&'static TextPool` 🤔